File: sea-shell-service-spring-web-flux/sea-shell-rest-service/src/main/java/org/jesperancinha/shell/webflux/service/fork/SeaShellPersonAccountRecursiveTask.java
package org.jesperancinha.shell.webflux.service.fork;

import lombok.AllArgsConstructor;
import lombok.Builder;
import org.jesperancinha.shell.webflux.data.SeaShellPersonDto;
import org.jesperancinha.shell.webflux.repo.ShellAccountRepository;
import org.jesperancinha.shell.webflux.service.SeaShellConverter;

import java.util.concurrent.RecursiveTask;

@AllArgsConstructor
@Builder
public class SeaShellPersonAccountRecursiveTask extends RecursiveTask<SeaShellPersonDto> {

    private final ShellAccountRepository accountRepository;
    private final SeaShellPersonDto seaShellPersonDto;

    @Override
    protected SeaShellPersonDto compute() {
        seaShellPersonDto.setAccountDto(SeaShellConverter
                .toAccountDto(accountRepository.findAccountByIdBlock(seaShellPersonDto.getAccountId())));
        return seaShellPersonDto;
    }
}

File: sea-shell-service-spring-web-flux/src/main/kotlin/org/jesperancinha/shell/webflux/service/fork/SeaShellPersonAccountRecursiveTask.kt
package org.jesperancinha.shell.webflux.service.fork

import lombok.AllArgsConstructor
import lombok.Builder
import org.jesperancinha.shell.webflux.data.SeaShellPersonDto
import org.jesperancinha.shell.webflux.repo.ShellAccountRepositoryImpl
import org.jesperancinha.shell.webflux.service.SeaShellConverter
import java.util.concurrent.RecursiveTask

@AllArgsConstructor
@Builder
class SeaShellPersonAccountRecursiveTask(
    private val accountRepository: ShellAccountRepositoryImpl,
    private val seaShellPersonDto: SeaShellPersonDto
) : RecursiveTask<SeaShellPersonDto>() {

    override fun compute(): SeaShellPersonDto = seaShellPersonDto.copy(
        accountDto = seaShellPersonDto.accountId
            ?.let { accountRepository.findAccountByIdBlock(it) }
            ?.let {
                SeaShellConverter.toAccountDto(it)
            }
    )
}
